### PR TITLE
OPS tool to assist in the rotation of SQL users

### PR DESCRIPTION
**Pull request by scottbommarito**
This job creates reader and writer SQL users on a SQL database and it's geo-replicated backup. It saves the username and password combinations in the folder that it runs in.

The users created have the structure `{NewPrincipalPrefix}{RoleName}xYYYYMMDD`.

Example:
```
CreateNewSqlUsers.exe 
    -PrimaryServerUrl "myprimaryserver.database.windows.net" 
    -PrimaryAdminUsername "myprimaryadmin" 
    -PrimaryAdminPassword "shhh-its-a-secret" 
    -SecondaryServerUrl "mysecondaryserver.database.windows.net" 
    -SecondaryAdminUsername "mysecondaryadmin" 
    -SecondaryAdminPassword "shhh-its-another-secret" 
    -NewPrincipalPrefix "MyUser" 
    -DatabaseName "databaseName" 
    -ReaderRole "my-reader-role" 
    -WriterRole "my-writer-role" 
    -Once
```

### CODE REVIEW

**Comment by chenriksson** [comment]
Do we have other Ops tools in NuGet.Jobs?

Wondering if we should have central place for all Ops tools, probably private repo (we have other Ops scripts under NuGetDeployment/Scripts). Also wondering if powershell scripts would be better for ops, as they're really easy to write and Azure powershell is usually pretty advanced.

**Comment by skofman1** [comment]
Thanks for the work @scottbommarito , however, the direction we need to take for automating secret rotation is Azure Automation: https://docs.microsoft.com/en-us/azure/key-vault/key-vault-key-rotation-log-monitoring 
This job is a good close gap, but I wouldn't like to submit it to our code base. Feel free to submit to Entropy repo.

**Comment by scottbommarito** [comment]
@skofman1 I wasn't sure where to put this but Entropy sounds like a good place. `NuGetDeployment/Scripts` is also good, as @chenriksson suggested. I was hoping that whoever would do the next secret rotation (before we have `Azure Automation`) could use this to speed up the process substantially.

**Comment by agr** [comment]
It probably should have been a powershell script...

**Comment by scottbommarito** [comment]
@agr I agree, but I'm more familiar with C#. To be honest I don't think it would be that hard to port it to PowerShell.

File: CreateNewSqlUsers/ArgumentNames.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace CreateNewSqlUsers
+{
+    public static class ArgumentNames
+    {
+        public const string PrimaryServerUrl = "PrimaryServerUrl";
```
Comment (agr) [inline CreateNewSqlUsers/ArgumentNames.cs]: It has "Url" in the name, but the tool expects a host name here, right?

Comment (scottbommarito) [reply CreateNewSqlUsers/ArgumentNames.cs]: It's something of the form
`_______.database.windows.net`
for DEV, the `____` starts with a d

Comment (agr) [reply CreateNewSqlUsers/ArgumentNames.cs]: yeah, a host name :)
Url starts with a schema.

**Pull request closed without merging**